common/src/sql/dbinit.sql

```diff
@@ -981,6 +989,15 @@ CREATE UNIQUE INDEX ON omicron.public.ip_pool (
 ) WHERE
     time_deleted IS NULL;
 
+/*
+ * Index ensuring uniqueness of IP pools by rack ID
+ */
+CREATE UNIQUE INDEX ON omicron.public.ip_pool (
```

Comment:
I'm not sure about the uniqueness constraint here, since that means we can only have one pool for Oxide services within any given rack. That might be fine, since there can be many ranges within that. It might also not matter right now.

Reply:
That was actually my intent - modulo https://github.com/oxidecomputer/omicron/issues/1276 , where this might move up to the availability zone layer.

I was viewing this as a pool of IPs for usage "by nexus for any external services" - and figured that I'd use the logic of roughly:
- For a service, where will it running (what rack / AZ)? -> Get a UUID
- For that UUID, look up the IP pool of available addresses
- For that pool, grab a free address

If that doesn't work, I could remove the `UNIQUE` constraint, but that's at least why the index exists. I didn't know of a use-case where we'd want multiple IP pools "reserved by the rack", but if we have one, I can relax this.

Reply:
Ok, so based on the changes to `populate.rs`, I'm inferring this is by design. There's no endpoint for controlling the Pool itself -- it's created when the control plane starts, and cannot be deleted -- only the ranges within it.

Reply:
Right - letting operators delete the IP pool which Nexus uses for external addresses seems like a footgun, so I didn't enable it.

Reply:
Definitely a footgun, it just wasn't clear initially that the pool lives as long as the database. Good idea!